### PR TITLE
Optionally preprocess segmentation maps for MobileViT

### PR DESCRIPTION
**Pull request by harisankar95**
# What does this PR do?
- Preprocessor can now accept segmentations maps as well and performs augmentations inline to input images.
- Tests added for preprocessing segmentation masks.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by harisankar95** [comment]
@amyeroberts Can you please review the PR? MobileViT image preprocessor is updated to accept segmentation maps inline to other preprocessors for segmentation models like that of Segformer.

File: src/transformers/models/mobilevitv2/modeling_mobilevitv2.py [hunk 1]
```diff
@@ -58,7 +58,7 @@
 
 
 MOBILEVITV2_PRETRAINED_MODEL_ARCHIVE_LIST = [
-    "apple/mobilevitv2-1.0-imagenet1k-256"
+    "apple/mobilevitv2-1.0-imagenet1k-256" "apple/mobilevitv2-1.0-voc-deeplabv3"
```
Comment (amyeroberts) [inline src/transformers/models/mobilevitv2/modeling_mobilevitv2.py:61]: These shouldn't be changed

File: src/transformers/models/mobilevit/image_processing_mobilevit.py [hunk 2]
```diff
@@ -306,6 +321,48 @@ def preprocess(
         ]
 
         data = {"pixel_values": images}
+
+        if segmentation_maps is not None:
```
Comment (amyeroberts) [inline src/transformers/models/mobilevit/image_processing_mobilevit.py:325]: We should add two new methods to keep in line with other models e.[g. SAM here](https://github.com/huggingface/transformers/blob/976189a6df796a2ff442dd81b022626c840d8c27/src/transformers/models/sam/image_processing_sam.py#L347): 
* preprocess_image
* preprocess_mask

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding this! 

Overall looks good - just some general comments about the structure. Could you add some tests to make sure the maps are process as expected?

File: src/transformers/models/mobilevitv2/modeling_mobilevitv2.py (hunk 1 above)
Comment (harisankar95) [reply src/transformers/models/mobilevitv2/modeling_mobilevitv2.py]: I have removed it in 1669db11280ecadd33b7535f92241b4877b15393

File: src/transformers/models/mobilevit/image_processing_mobilevit.py (hunk 2 above)
Comment (harisankar95) [reply src/transformers/models/mobilevit/image_processing_mobilevit.py:411]: Added additional methods to preprocess images and masks seperately in 877b81a4f6a34c24b5156acb19244bb3c8c83663

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this and the tests - looks great!

**Comment by harisankar95** [comment]
I have corrected the error in the tests_torch.  I will wait for the https://github.com/huggingface/transformers/pull/28432 to be merged to fix the remaining CI tests

File: src/transformers/models/mobilevitv2/modeling_mobilevitv2.py [hunk 3]
```diff
@@ -980,8 +980,8 @@ def forward(
         >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
         >>> image = Image.open(requests.get(url, stream=True).raw)
 
-        >>> image_processor = AutoImageProcessor.from_pretrained("apple/mobilevitv2-1.0-imagenet1k-256")
-        >>> model = MobileViTV2ForSemanticSegmentation.from_pretrained("apple/mobilevitv2-1.0-imagenet1k-256")
+        >>> image_processor = AutoImageProcessor.from_pretrained("apple/mobilevitv2-1.0-voc-deeplabv3")
+        >>> model = MobileViTV2ForSemanticSegmentation.from_pretrained("apple/mobilevitv2-1.0-voc-deeplabv3")
```
Comment (amyeroberts) [inline src/transformers/models/mobilevitv2/modeling_mobilevitv2.py:984]: These should remain the same

**Review by amyeroberts — APPROVED** [review]
Thanks for adding and iterating on this! 

Just a small nit about keeping the same checkpoints in the doc example. Otherwise LGTM!

File: src/transformers/models/mobilevitv2/modeling_mobilevitv2.py (hunk 3 above)
Comment (harisankar95) [reply src/transformers/models/mobilevitv2/modeling_mobilevitv2.py]: Thanks for letting me know. I have changed the checkpoints back to the initial one.

**Pull request merged**